Odata-docs/webapi/pagesize-top-maxtop.md

```diff
@@ -0,0 +1,87 @@
+---
+title: "PageSize, Top and MaxTop"
+description: "This tutorial describes how to configure PageSize, Top and MaxTop."
+
+author: KenitoInc
+ms.author: kemunga
+
+ms.date: 3/1/2022
+---
+# PageSize, Top and MaxTop
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v6.md)]
+
+In OData WebApi we support both client-driven and server-driven paginations.
+
+### Client-driven pagination
+In client-driven paging, we request the server to return the specified number of results. There is no `nextLink` that is returned.
+
+The OData WebApi deals with client-driven paging using `$skip` and `$top` query options.
+
+The `$top` query option requests the number of items in the queried collection to be included in the result.
+
+The `$skip` query option requests the number of items in the queried collection that are to be skipped and not included in the result.
+
+The `MaxTop()` sets the maximum value that can be set for the `$top` query parameter. It does not directly limit the number of items returned per request.
+
+In `Microsoft.AspNetCore.OData v7.x`, we configure `MaxTop()` as follows:
+
+```csharp
+app.UseMvc(routeBuilder =>
+{
+    routeBuilder.Select().Expand().Filter().OrderBy().MaxTop(100).Count();
+
+    routeBuilder.MapODataServiceRoute("ODataRoute", "odata", builder.GetEdmModel());
+});
+```
+
+`MaxTop()` only impacts the scenarios in which the request Uri contains `$top`. If the `$top` value exceeds the `MaxTop` value, you will get an error message.
+
+For example, if we set `MaxTop` value to 20 and then pass `$top` query parameter value as 100, you will get the error message as shown below:
```

Comment:
```suggestion
For example, if we set `MaxTop` value to 100 and then pass `$top` query parameter value as 150 , you will get the error message as shown below:
```
Keep it consistent for people who may skim over the docs and are just looking at the code snippets.